### PR TITLE
BFD-3184: Add lombok annotations to bfd-server-war module classes

### PR DESCRIPTION
**Pull request by rcrupp**
**JIRA Ticket:**
[BFD-3184](https://jira.cms.gov/browse/BFD-3184)

**User Story or Bug Summary:**

As a BFD developer, I would like to consistently use lombok annotations in order to avoid writing redundant getter and setter methods for data classes. 

---

### What Does This PR Do?
* Adds Lombok `@Getter`, `@Setter`, and `@AllArgsConstructor` annotations where appropriate.
* Updates six classes in the `bfd-server-war` module.
* Updates one class in the `bfd-shared-test-utils` module.

### What Should Reviewers Watch For?
If you're reviewing this PR, please check for these things in particular:
* Verify all PR security questions and checklists have been completed and addressed.

### What Security Implications Does This PR Have?

Submitters should complete the following questionnaire:

* If the answer to any of the questions below is **Yes**, then **you must** supply a link to the associated Security Impact Assessment (SIA), security checklist, or other similar document in Confluence here: **N/A**

    * Does this PR add any new software dependencies? 
      * [ ] Yes
      * [X] No
    * Does this PR modify or invalidate any of our security controls?
      * [ ] Yes
      * [X] No
    * Does this PR store or transmit data that was not stored or transmitted before?
      * [ ] Yes
      * [X] No

* If the answer to any of the questions below is **Yes**, then please add @StewGoin as a reviewer, and note that this PR **should not be merged** unless/until he also approves it.
    * Do you think this PR requires additional review of its security implications for other reasons?
      * [ ] Yes
      * [X] No

### What Needs to Be Merged and Deployed Before this PR?

This PR cannot be either merged or deployed until the following prerequisite changes have been fully deployed:

* N/A

### Submitter Checklist
I have gone through and verified that...:

* [X] I have named this PR and branch so they are [automatically linked](https://confluence.atlassian.com/adminjiracloud/integrating-with-development-tools-776636216.html) to the (most) relevant Jira issue. Ie: `BFD-123: Adds foo`
* [X] This PR is reasonably limited in scope, to help ensure that:
    1. It doesn't unnecessarily tie a bunch of disparate features, fixes, refactorings, etc. together.
    2. There isn't too much of a burden on reviewers.
    3. Any problems it causes have a small "blast radius".
    4. It'll be easier to rollback if that becomes necessary.
* [X] This PR includes any required documentation changes, including `README` updates and changelog / release notes entries.
* [X] The data dictionary has been updated with any field mapping changes, if any were made.
* [X] All new and modified code is appropriately commented, such that the what and why of its design would be reasonably clear to engineers, preferably ones unfamiliar with the project.
* [X] All tech debt and/or shortcomings introduced by this PR are detailed in `TODO` and/or `FIXME` comments, which include a JIRA ticket ID for any items that require urgent attention.
* [X] Reviews are requested from both:
    * At least two other engineers on this project, at least one of whom is a senior engineer or owns the relevant component(s) here.
    * Any relevant engineers on other projects (e.g. DC GEO, BB2, etc.).
* [X] Any deviations from the other policies in the [DASG Engineering Standards](https://github.com/CMSgov/cms-oeda-dasg/blob/master/policies/engineering_standards.md) are specifically called out in this PR, above.
    * Please review the standards every few months to ensure you're familiar with them.

### CODE REVIEW

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/LoadedFilterManager.java [hunk 1]
```diff
@@ -37,7 +39,7 @@ public class LoadedFilterManager {
   private EntityManager entityManager;
 
   /** The filter set. */
-  private List<LoadedFileFilter> filters;
+  @Getter private List<LoadedFileFilter> filters;
```
Comment (meliGuzman) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/LoadedFilterManager.java:42]: @rcrupp  I've used lombok before but I'm not familiar with seeing an @ Getter get used like this. What's the reasoning?

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/CommonCodings.java [hunk 2]
```diff
@@ -1,6 +1,11 @@
 package gov.cms.bfd.server.war.commons;
```
Comment (meliGuzman) [inline apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/CommonCodings.java:2]: @rcrupp this is just a general comment. Could you additional tests just to verify if there's no test coverage for these getters / where the class gets instantiated

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/LoadedFilterManager.java (hunk 1 above)
Comment (brianburton) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/LoadedFilterManager.java:42]: The annotation can be applied at either the class or method level.  At the class level it exposes every field with a getter.  At the field level it adds a getter for just that field.  In this case the class only had a getter for `filters` so we don't want to accidentally add one for `entityManager`.  Adding the annotation to `filters` directly preserves the existing encapsulation of `entityManager`.

**Review by brianburton — APPROVED** [review]
Nice improvements.

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/LoadedFilterManager.java (hunk 1 above)
Comment (meliGuzman) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/LoadedFilterManager.java:42]: Oh!! Thank you for explaining @brianburton !

**Review by brandoncruz3 — APPROVED** [review]
Nice!

File: apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/CommonCodings.java (hunk 2 above)
Comment (mjburling) [reply apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/commons/CommonCodings.java:2]: 👋 Hi, @meliGuzman. We can address you questions outside of this PR if you should still have any. Thanks!

**Pull request merged**